src/devices/Mcp25xxx/Register/MessageReceive/RxBxSidl.cs

```diff
@@ -0,0 +1,158 @@
+    /// </summary>
+    public class RxBxSidl : IRegister
+    {
+        /// <summary>
+        ///  Initializes a new instance of the RxBxSidl class.
+        /// </summary>
+        /// <param name="rxBufferNumber">Receive Buffer Number.</param>
+        /// <param name="eid">
+        /// Extended Identifier bits.
+        /// These bits contain the three Least Significant bits of the Standard Identifier for the received message.
+        /// </param>
+        /// <param name="ide">
+        /// Extended Identifier Flag bit.
+        /// This bit indicates whether the received message was a standard or an extended frame.
+        /// True = Received message was an extended frame.
+        /// False = Received message was a standard frame.
+        /// </param>
+        /// <param name="srr">
+        /// Standard Frame Remote Transmit Request bit (valid only if the IDE bit = 0).
+        /// True = Standard frame Remote Transmit Request received.
+        /// False = Standard data frame received.
+        /// </param>
+        /// <param name="sid">
+        /// Extended Identifier bits.
+        /// </param>
+        public RxBxSidl(RxBufferNumber rxBufferNumber, byte eid, bool ide, bool srr, byte sid)
+        {
+            if (eid > 3)
+            {
+                throw new ArgumentException($"Invalid EID value {eid}.", nameof(eid));
+            }
+
+            if (sid > 7)
+            {
+                throw new ArgumentException($"Invalid SID value {sid}.", nameof(sid));
+            }
+
+            RxBufferNumber = rxBufferNumber;
+            Eid = eid;
+            Ide = ide;
+            Srr = srr;
+            Sid = sid;
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the RxBxSidl class.
+        /// </summary>
+        /// <param name="value">The value that represents the register contents.</param>
+        public RxBxSidl(RxBufferNumber rxBufferNumber, byte value)
+        {
+            RxBufferNumber = rxBufferNumber;
+            Eid = (byte)(value & 0b0000_0011);
+            Ide = (value & 0b0000_1000) == 0b0000_1000;
+            Srr = (value & 0b0001_0000) == 0b0001_0000;
+            Sid = (byte)((value & 0b1110_0000) >> 5);
+        }
+
+        /// <summary>
+        /// Receive Buffer Number.
+        /// </summary>
+        public RxBufferNumber RxBufferNumber { get; }
+
+        /// <summary>
+        /// Extended Identifier bits.
+        /// These bits contain the three Least Significant bits of the Standard Identifier for the received message.
+        /// </summary>
+        public byte Eid { get; }
+
+        /// <summary>
+        /// Extended Identifier Flag bit.
+        /// This bit indicates whether the received message was a standard or an extended frame.
+        /// True = Received message was an extended frame.
+        /// False = Received message was a standard frame.
+        /// </summary>
+        public bool Ide { get; }
```

Comment:
This is called EFF (29-bit identifier) in SocketCAN and other places I've seen (standard/11bit identifiers are called SFF).

Reply:
please read the rest of the comments in this class